libr/core/cmd_type.c

```diff
@@ -311,31 +312,57 @@ static int typelist(void *p, const char *k, const char *v) {
 	return 1;
 }
 
-static int sdbforcb (void *p, const char *k, const char *v) {
-	if (!strncmp (v, "type", strlen ("type") + 1)) {
-		r_cons_println (k);
-	}
+static int sdbforcb_default (void *p, const char *k, const char *v) {
+	r_cons_println (k);
+	return 1;
+}
+
+static int sdbforcb_json (void *p, const char *k, const char *v) {
+	Sdb *sdb = (Sdb *)p;
+	char *sizecmd = malloc (strlen("type..size") + strlen(k) + 1);
+	char *formatcmd = malloc (strlen("type.") + strlen(k) + 1);
+	sprintf (sizecmd, "type.%s.size", k);
+	sprintf (formatcmd, "type.%s", k);
+	r_cons_printf ("{\"type\":\"%s\",\"size\":%d,\"format\":\"%c\"}", k,
+			atoi(sdb_querys(sdb, NULL, -1, sizecmd)),
+			sdb_querys(sdb, NULL, -1, formatcmd)[0]);
```

Comment:
Are all formats 1-character only? Is that a valid assumption?

Reply:
dont think so, a char* should be valid

Reply:
problem was sdb_querys returns a string ending with `\n`. I don't know how to trim it — without writing too much.

Reply:
oh ...r_str_trim.